### ISSUE TITLE
Clicking again does not desactivate the on screen keyboard

### ISSUE THREAD

**Issue by Gogny** [Bradan/Gnome-On-Screen-Keyboard-Button#6]
Hello !
I have a Lenovo C740, running Pop!_OS 20.10. To desactivate the on screen keyboard after having enabled it with this handy button, I have to manually turn the screen keyboard off in the accessibility settings. 
That is not a disaster but is definitively something to improve, this extension is so useful !

Thank you for reading me

**Comment by purpleidea** [comment]
I can confirm the same behaviour on GNOME Shell 3.38.4 on Fedora 33.

If someone knows the fix, that would be awesome!

**Comment by Bradan** [comment]
Sorry I missed this issue in 2020 and the answer got me caught on it again.

It is intended to work like it does. The main reason this extension exists is because the keyboard doesn't show up automatically in programs like Google Chrome if the address bar is focused (probably because it doesn't use GTK or draws the widgets itself?). However I can understand that you might not want it not to show up automatically at all.

If I will ever implement it, I would make it an optional feature, but the complexity of this whole partially undocumented and frequently changing API makes it unbelievably hard to maintain any extension. I mean look at the code, it is not really large and even though it only uses a hand full of basic functionalities I had problems between versions.

I will let the issue stay open, however I will not promise anything. If someone has a not too large pull request I may include it, but I want it to be simple and maintainable.

**Comment by purpleidea** [comment]
> It is intended to work like it does.

No, there's actually a bug-- the toggle is supposed to enable/disable the keyboard from being _allowed_ to appear-- it's not that it's not hiding it properly, it's that it's not disabling the toggle. The working toggle can be found in the accessibility menu. The extension works fine to toggle it on, but not to toggle it off.

As for hiding.showing the keyboard, that happens based on whether a text input has focus, and if the person pressed the temporary hide button that's visible in the keyboard.

HTH nice extension, I just don't know the every changing API well enough to know the correct patch. Maybe I'll dig into it next week.

**Comment by Bradan** [comment]
> No, there's actually a bug-- the toggle is supposed to enable/disable the keyboard from being _allowed_ to appear

No it is not. That's what I just said. I want the OSK to stay enabled in the accessibility settings. I just enforced it to be enabled in case the user forgot to do so before, because the extension doesn't make sense without it being enabled. Not a bug, but a feature request.

If you implement it, it must be optional. With a new settings menu of the extension maybe.
